pages/common/uuencode.md

```diff
@@ -0,0 +1,16 @@
+# uuencode
+
+> Encode binary files into ASCII for transport via mediums that only support simple ASCII encoding.
+> More information: <https://manned.org/uuencode>.
+
+- Encode a file and print the result to stdout:
+
+`uuencode {{path/to/input_file}} {{output_file_name_after_decoding}}`
```

Comment:
Why specify a output file name if you print it to stdout?

Reply:
Yeah, it's confusing.  `output_file_name_after_decoding` is the name of the file that will be created when I decode the file.  Because `uuencode` doesn't just encode the file contents; it also encodes the file permissions (`644` in the example below) and the name.

```console
$ echo hello > myfile

$ cat myfile
hello

$ uuencode myfile decodedfile > encoded

$ ls
encoded  myfile

$ cat encoded
begin 644 decodedfile
&:&5L;&\*
`
end

$ uudecode encoded

$ ls
decodedfile  encoded  myfile

$ cat decodedfile
hello


```

Reply:
That's really confusing

Reply:
Yea, I'm not sure if it's the tool itself or just the stupid BSD version, but it's definitely **hard** to use, as least for a beginner.  I had to try a bunch of times before I could even get it working 😝

Reply:
Sounds like there should be an alternative for it.

Reply:
Yeah.
- https://github.com/murtyjones/uuencode
- https://github.com/zacbarton/node-uuencode
- https://github.com/bnjf/compre.sh
- https://github.com/cention-sany/uuencode
- https://en.wikipedia.org/wiki/Uuencoding